app/src/main/java/com/infomaniak/mail/ui/main/thread/PrintMailFragment.kt

```diff
@@ -0,0 +1,100 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.main.thread
+
+import android.content.Context
+import android.os.Bundle
+import android.print.PrintManager
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebView
+import androidx.core.view.get
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.infomaniak.mail.R
+import com.infomaniak.mail.data.LocalSettings
+import com.infomaniak.mail.databinding.FragmentPrintMailBinding
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class PrintMailFragment : Fragment() {
+
+    @Inject
+    lateinit var localSettings: LocalSettings
+
+    private var _binding: FragmentPrintMailBinding? = null
+    private val binding get() = _binding!! // This property is only valid between onCreateView and onDestroyView
+
+    private val navigationArgs: PrintMailFragmentArgs by navArgs()
+    private val threadViewModel: ThreadViewModel by viewModels()
+    private val threadAdapter inline get() = binding.messagesList.adapter as ThreadAdapter
+
+    private var subject = ""
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return FragmentPrintMailBinding.inflate(inflater, container, false).also { _binding = it }.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?): Unit = with(threadViewModel) {
+        super.onViewCreated(view, savedInstanceState)
+        setupAdapter()
+        threadLive.observe(viewLifecycleOwner) { thread ->
+            thread?.subject?.let { subject = it }
+        }
+        messagesLive.observe(viewLifecycleOwner) { messages ->
+            threadAdapter.submitList(messages.filter { it.uid == navigationArgs.messageUid })
+        }
+        navigationArgs.openThreadUid?.let {
+            reassignThreadLive(it)
+            reassignMessagesLive(it)
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onPause() {
+        super.onPause()
+        findNavController().popBackStack()
+    }
```

Comment:
Why did you need this?

Reply:
To avoid going back to PrintMailFragment after printing

Reply:
So leaving while it's loading gets you out as well then?